var/spack/repos/builtin/packages/delphes/package.py

```diff
@@ -38,25 +39,18 @@ class Delphes(CMakePackage):
     version('3.0.6', sha256='9e225731d57d2a76d35886841f8eff121bb3a45560b16077bd8c351151581d88')
     version('3.0.5', sha256='ab64ec6d2476fbfa40562e7edb510a8ab4c4fe5be77a4353ebf315c2af181a80')
 
-    depends_on('cmake', type='build')
-    depends_on('root cxxstd=14', when='cxxstd=14')
-    depends_on('root cxxstd=17', when='cxxstd=17')
+    variant('pythia8', default=True,
+            description="build with pythia8")
 
-    variant('build_type', default='Release',
-            description='The build type to build',
-            values=('Debug', 'Release'))
-
-    variant('cxxstd',
-            default='17',
-            values=('14', '17'),
-            multi=False,
-            description='Use the specified C++ standard when building.')
+    depends_on('root')
+    depends_on('pythia8', when="+pythia8")
 
     def cmake_args(self):
         args = []
         # C++ Standard
         args.append('-DCMAKE_CXX_STANDARD=%s'
-                    % self.spec.variants['cxxstd'].value)
```

Comment:
It looks like you removed the `depends_on(root cxxstd=14, when=cxxstd=14)` constraints so to simplify the concretization. I would recommend against that (although this decision is not strictly a blocker) unless you were having concretization issues: if other packages were to depend on `delphes`, they would also need to know to check `root` for the `cxxstd`.

Reply:
That's a good question -- I generally prefer this way of setting the cxxstd for root based projects, because it is what is actually generally recommended: use the standard that root uses. There is no strict dependency as far as I know, but there are some cases where builds can fail. I'm not sure how either will help for packages depending on delphes, they will still need to set up the proper cxxstd for their build?

I'll leave ithe existing version for now, and try out a few test builds to review this, because it concerns any package that depends on root.